### PR TITLE
Adds arccosh alias for acosh and adds an alias consistency test

### PR DESCRIPTION
**Pull request by mruberry**
This adds the torch.arccosh alias and updates alias testing to validate the consistency of the aliased and original operations. The alias testing is also updated to run on CPU and CUDA, which revealed a memory leak when tracing (see https://github.com/pytorch/pytorch/issues/43119).

### CODE REVIEW

File: test/test_op_aliases.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+              lambda d: torch.randn(20, device=d), get_args=lambda d: (.4, .6)),
+    AliasInfo('linalg.det', torch.linalg.det, 'det', torch.det,
+              lambda d: torch.randn(10, 10, device=d),
+              decorators=(skipCPUIfNoLapack, skipCUDAIfNoMagma)),
+    # NOTE: only runs on CPU because it leaks CUDA memory
+    #   (see https://github.com/pytorch/pytorch/issues/43119)
+    AliasInfo('outer', torch.outer, 'ger', torch.ger,
+              lambda d: torch.randn(20, device=d), get_args=lambda d: (torch.randn(20, device=d),),
+              decorators=(onlyCPU,)),
+    AliasInfo('arccosh', torch.arccosh, 'acosh', torch.acosh,
```
Comment (ngimel) [inline test/test_op_aliases.py:51]: it's not important, but are you sure you want to test almost only nans here?

File: test/test_op_aliases.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        def _test_alias_computation(self, device, info=info):
+            alias_op = info.alias_op
+            original_op = info.original_op
+
+            inp = info.get_input(device)
+            args = info.get_args(device)
+            alias_result = alias_op(inp.clone(), *args)
+            original_result = alias_op(inp.clone(), *args)
+
+            self.assertEqual(alias_result, original_result)
```
Comment (ngimel) [inline test/test_op_aliases.py:130]: set tolerances to 0?

File: test/test_op_aliases.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        self.alias_op = alias_op
+        self.original_name = original_name
+        self.original_op = original_op
+        self.get_input = get_input
+        self.get_args = get_args
+        self.decorators = decorators
+
+alias_infos = (
+    AliasInfo('absolute', torch.absolute, 'abs', torch.abs,
+              lambda d: torch.randn(20, device=d) - .5),
```
[the remark this answers is not in the document]
Comment (mruberry) [reply test/test_op_aliases.py:36]: you're right, it's unneeded. I must have thought randn was sampling from 0 to 1, not with mean zero.

File: test/test_op_aliases.py (hunk 1 above)
Comment (mruberry) [reply test/test_op_aliases.py:51]: I was also thinking about this. I'll just add 2 to the input tensor so we're in the domain

File: test/test_op_aliases.py (hunk 2 above)
Comment (mruberry) [reply test/test_op_aliases.py:130]: Excellent idea

File: test/test_op_aliases.py (hunk 1 above)
Comment (ngimel) [reply test/test_op_aliases.py:51]: mostly in the domain :-)

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**